mods/default/torch.lua

```diff
@@ -3,152 +3,116 @@
+		local nodedef = minetest.registered_nodes[newnode.name]
+		if not (nodedef and nodedef.groups and
+				nodedef.groups.igniter and nodedef.groups.igniter > 0) then
+			minetest.sound_play("default_cool_lava",
+				{pos = pos, max_hear_distance = 16, gain = 0.1})
+		end
+		-- Remove the torch node
+		return false
 	end
-	-- Remove the torch node
-	return false
-end
-
-minetest.register_node("default:torch", {
-	description = S("Torch"),
-	drawtype = "mesh",
-	mesh = "torch_floor.obj",
-	inventory_image = "default_torch_on_floor.png",
-	wield_image = "default_torch_on_floor.png",
-	tiles = {{
-		    name = "default_torch_on_floor_animated.png",
-		    animation = {type = "vertical_frames", aspect_w = 16, aspect_h = 16, length = 3.3}
-	}},
-	paramtype = "light",
-	paramtype2 = "wallmounted",
-	sunlight_propagates = true,
-	walkable = false,
-	liquids_pointable = false,
-	light_source = 12,
-	groups = {choppy=2, dig_immediate=3, flammable=1, attached_node=1, torch=1},
-	drop = "default:torch",
-	selection_box = {
-		type = "wallmounted",
-		wall_bottom = {-1/8, -1/2, -1/8, 1/8, 2/16, 1/8},
-	},
-	sounds = default.node_sound_wood_defaults(),
-	on_place = function(itemstack, placer, pointed_thing)
+	def.on_place = function(itemstack, placer, pointed_thing)
 		local under = pointed_thing.under
 		local node = minetest.get_node(under)
-		local def = minetest.registered_nodes[node.name]
-		if def and def.on_rightclick and
+		local node_def = minetest.registered_nodes[node.name]
+		if node_def and node_def.on_rightclick and
 			not (placer and placer:is_player() and
 			placer:get_player_control().sneak) then
-			return def.on_rightclick(under, node, placer, itemstack,
+			return node_def.on_rightclick(under, node, placer, itemstack,
 				pointed_thing) or itemstack
 		end
-
 		local above = pointed_thing.above
 		local wdir = minetest.dir_to_wallmounted(vector.subtract(under, above))
-		local fakestack = itemstack
 		if wdir == 0 then
-			fakestack:set_name("default:torch_ceiling")
+			itemstack:set_name(name .. "_ceiling")
 		elseif wdir == 1 then
-			fakestack:set_name("default:torch")
+			itemstack:set_name(name)
 		else
-			fakestack:set_name("default:torch_wall")
+			itemstack:set_name(name .. "_wall")
 		end
+		itemstack = minetest.item_place(itemstack, placer, pointed_thing, wdir)
+		itemstack:set_name(name)
+		return itemstack
+	end
 
-		itemstack = minetest.item_place(fakestack, placer, pointed_thing, wdir)
-		itemstack:set_name("default:torch")
+	def.mesh = "torch_floor.obj"
+	def.selection_box = {
+		type = "wallmounted",
+		wall_bottom = {-1/8, -1/2, -1/8, 1/8, 2/16, 1/8}
+	}
+	minetest.register_node(":" .. name, def)
 
-		return itemstack
-	end,
-	floodable = true,
-	on_flood = on_flood,
-	on_rotate = false
-})
+	local torch_wall = table.copy(def)
+	torch_wall.mesh = "torch_wall.obj"
+	torch_wall.selection_box = {
+		type = "wallmounted",
+		wall_side = {-1/2, -1/2, -1/8, -1/8, 1/8, 1/8}
+	}
+	torch_wall.groups.not_in_creative_inventory = 1
```

Comment:
since groups is optional according to the documentation, there needs to be `if torch_wall.groups == nil then torch_wall.groups = {} end` here

Reply:
Maybe better is `def.groups = def.groups or {}`?

Reply:
Sure.